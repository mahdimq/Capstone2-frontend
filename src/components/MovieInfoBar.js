import React from 'react';
import PropTypes from 'prop-types';
import { calculateTime, formatDate } from '../helpers/helpers';

// Style Imports
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faClock, faCalendarAlt } from '@fortawesome/free-solid-svg-icons';
import { StyledMovieInfoBar } from '../styles/StyledMovieInfoBar';

function MovieInfoBar({ time, released, budget }) {
	return (
		<StyledMovieInfoBar>
			{/* MOVIE RUN TIME */}
			<div className='movieinfobar-content'>
				<div className='movieinfobar-content-col'>
					<span className='movieinfobar-info'>
						<FontAwesomeIcon
							className='icon'
							style={{ color: 'orange', marginRight: '0.4em' }}
							icon={faClock}
						/>
						Running time: {calculateTime(time)}
					</span>
				</div>

				{/* RELEASE DATE  */}
				<div className='movieinfobar-content-col'>
					<span className='movieinfobar-info'>
						<FontAwesomeIcon
							className='icon'
							style={{ color: 'skyBlue', marginRight: '0.4em' }}
							icon={faCalendarAlt}
						/>
						Released: {formatDate(released)}
					</span>
				</div>
			</div>
		</StyledMovieInfoBar>
	);
}

MovieInfoBar.propTypes = {
	time: PropTypes.number
};

export default MovieInfoBar;
