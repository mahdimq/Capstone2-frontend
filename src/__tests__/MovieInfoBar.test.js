import React from 'react';
import { render } from '@testing-library/react';
import MovieInfoBar from '../components/MovieInfoBar';

it('renders without crashing', function () {
	render(<MovieInfoBar />);
});

it('matches snapshot', function () {
	const { asFragment } = render(<MovieInfoBar />);
	expect(asFragment()).toMatchSnapshot();
});
