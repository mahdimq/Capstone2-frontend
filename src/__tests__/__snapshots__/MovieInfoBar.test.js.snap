// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ lmgoIb"
  >
    <div
      class="movieinfobar-content"
    >
      <div
        class="movieinfobar-content-col"
      >
        <span
          class="movieinfobar-info"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-clock fa-w-16 icon"
            data-icon="clock"
            data-prefix="fas"
            focusable="false"
            role="img"
            style="color: orange; margin-right: 0.4em;"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256,8C119,8,8,119,8,256S119,504,256,504,504,393,504,256,393,8,256,8Zm92.49,313h0l-20,25a16,16,0,0,1-22.49,2.5h0l-67-49.72a40,40,0,0,1-15-31.23V112a16,16,0,0,1,16-16h32a16,16,0,0,1,16,16V256l58,42.5A16,16,0,0,1,348.49,321Z"
              fill="currentColor"
            />
          </svg>
          Running time: NaNh NaNm
        </span>
      </div>
      <div
        class="movieinfobar-content-col"
      >
        <span
          class="movieinfobar-info"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-calendar-alt fa-w-14 icon"
            data-icon="calendar-alt"
            data-prefix="fas"
            focusable="false"
            role="img"
            style="color: skyBlue; margin-right: 0.4em;"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V192H0v272zm320-196c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zm0 128c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zM192 268c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zm0 128c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zM64 268c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12H76c-6.6 0-12-5.4-12-12v-40zm0 128c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12H76c-6.6 0-12-5.4-12-12v-40zM400 64h-48V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H160V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H48C21.5 64 0 85.5 0 112v48h448v-48c0-26.5-21.5-48-48-48z"
              fill="currentColor"
            />
          </svg>
          Released: Invalid Date
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
